### PR TITLE
contrib/jackc/pgx/v5: add initial support

### LINKED ISSUE #697: proposal: integrate with github.com/jackc/pgx
How to integrate with https://github.com/jackc/pgx ?

### PR DESCRIPTION
**Pull request by j-sv**
### What does this PR do?

Like #1840, this is based on #1537 in an attempt to get support for this merged. It implements all the tracer interfaces in `jackc/pgx/v5` in addition to the `QueryTracer`. It adds unit tests and examples for the connection and pool usage.

Tracers other than `QueryTracer` are disabled by default but can be enabled by options.

This PR should close #697.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Comment by j-sv** [comment]
@zarirhamza, of course!

File: contrib/jackc/pgx/v5/example_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	// The package exposes the same connect functions as the regular pgx library
+	// which sets up a tracer and connects as usual.
+	db, err := pgxtracer.Connect(ctx, "postgres://pqgotest:password@localhost/pqgotest?sslmode=disable")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close(ctx)
+
+	// Any calls made to the database will be traced as expected.
+	rows, err := db.Query(ctx, "SELECT name FROM users WHERE age=?", 27)
```
Comment (ajgajg1134) [inline contrib/jackc/pgx/v5/example_test.go:24]:
```suggestion
	rows, err := db.Query(ctx, "SELECT name FROM users WHERE age=$1", 27)
```

File: contrib/jackc/pgx/v5/options.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package pgx
+
+type Option func(*tracer)
+
+func WithServiceName(name string) Option {
```
Comment (ajgajg1134) [inline contrib/jackc/pgx/v5/options.go:5]: Let's add godoc comments to all these public functions

File: contrib/jackc/pgx/v5/trace.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+	finish(ctx, data.Err)
+}
+
+func (t *tracer) TraceConnectStart(ctx context.Context, _ pgx.TraceConnectStartData) context.Context {
+	if !t.traceConnect {
+		return ctx
+	}
+
+	opts := t.spanOptions(
```
Comment (ajgajg1134) [inline contrib/jackc/pgx/v5/trace.go:169]: The now time is the default so no need to create and use this option here

File: contrib/jackc/pgx/v5/pgxpool/pgxpool.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+func New(ctx context.Context, connString string, opts ...pgxtracer.Option) (*pgxpool.Pool, error) {
+	config, err := pgxpool.ParseConfig(connString)
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (ajgajg1134) [inline contrib/jackc/pgx/v5/pgxpool/pgxpool.go:16]:
```suggestion
```

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
This is just a very initial review, @zarirhamza and @rarguellof will give a more complete review next week. Thanks again for opening this contribution!!

File: contrib/jackc/pgx/v5/options.go (hunk 2 above)
Comment (j-sv) [reply contrib/jackc/pgx/v5/options.go:6]: Updated with comments and tried to add some examples.

File: contrib/jackc/pgx/v5/pgxpool/pgxpool.go (hunk 4 above)
Comment (j-sv) [reply contrib/jackc/pgx/v5/pgxpool/pgxpool.go]: Updated.

File: contrib/jackc/pgx/v5/trace.go (hunk 3 above)
Comment (j-sv) [reply contrib/jackc/pgx/v5/trace.go]: Removed the start time from all span options.

**Comment by sejin-P** [comment]
@ajgajg1134 Thank you for your attention, how's it coming? 

Additionally, I appreciate to your contribution, @j-sv

**Comment by zarirhamza** [comment]
Just fixing some things up. Will have more updates later in the week

**Comment by zarirhamza** [comment]
Just an update for everyone watching, we decided to continue this on another PR #2236 because it required a bit more work to be more complete especially on the testing side.

We will continue using the bulk of the tracing capabilities from this PR which was a massive contribution, so thank you!

**Comment by katiehockman** [comment]
@zarirhamza should we just go ahead and close this PR then, to clean up the backlog a bit?

**Comment by zarirhamza** [comment]
Moved to separate PR so closing this one.

**Pull request closed without merging**